Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -951,24 +953,18 @@ bool CompareWorkspaces::checkRunProperties(const API::Run &run1,
     g_log.debug() << "WS2 number of logs: " << ws2logs.size() << "\n";
     recordMismatch("Different numbers of logs");
     return false;
-  }
-
-  // Now loop over the individual logs
-  bool matched(true);
-  int64_t length(static_cast<int64_t>(ws1logs.size()));
-  PARALLEL_FOR_IF(true)
-  for (int64_t i = 0; i < length; ++i) {
-    PARALLEL_START_INTERUPT_REGION
-    if (matched) {
+  } else {
+    // Now loop over the individual logs
+    for (size_t i = 0; i < ws1logs.size(); ++i) {
       if (*(ws1logs[i]) != *(ws2logs[i])) {
-        matched = false;
+        g_log.debug("WS1 log: " + ws1logs[i]->name());
```

Comment:
Can I suggest adding a if(g_log.is(Logger.DEBUG))` and only doing the debug logging if the debug log is set? Logging is quite expensive, especially in a loop, so it would be good to avoid any extra work in a tight loop.

Reply:
:+1: